src/context/mapStateSlice.ts

```diff
@@ -1,30 +1,33 @@
-import { Map, Set } from 'immutable';
+import { Map } from 'immutable';
 import { createSlice, PayloadAction } from '@reduxjs/toolkit';
 
 import { RootState } from './store';
-import { LayerType } from '../config/types';
+import { LayersMap, LayerType } from '../config/types';
 
-interface LayersSet extends Set<LayerType> {}
 interface DateRange {
   startDate?: number;
   endDate?: number;
 }
 interface MapState extends Map<string, any> {}
 
 const initialState: MapState = Map({
-  layers: Set<LayerType>(),
-  dateRange: {},
+  layers: Map() as LayersMap,
+  dateRange: {} as DateRange,
```

Comment:
Again, don't mix and match Immutable & regular JS objects. Since this is all Redux state with Immutable, we'd like `DateRange` (and `MapState` for that matter) to be strongly typed `Maps` instead of objects or weak (because of the `any`) types.

[This SO post is useful for that](https://stackoverflow.com/questions/46284131/typescript-and-immutable-js-interface) - read the first answer.

You can do something like:

```
// The default Map interface accepts <K,V>: Key, Value.
// Build an interface that also accepts 'T': the shape of your data.
// You might want to add other methods, e.g. `update`? Probably can cut and paste more or less from
// https://github.com/immutable-js/immutable-js/blob/7f4e61601d92fc874c99ccf7734d6f33239cec8c/type-definitions/Immutable.d.ts#L2449
export interface IImmutableMap<T, K, V> extends Map<K, V> {
  toJS(): T;
  get<I extends keyof T>(key: I & K): T[I] & V;
  set<S extends keyof T>(key: S & K, value: T[S] & V): Map<K, V>;
}


export interface IMapState extends Map<string, any> {
  layers: LayersMap,
  // Do a similar typed definition for `DateRange`
  dateRange: DateRange,
};

export type MapState = IImmutableMap<IMapState, string, any>;

// Update the type definition on initial state to your type.
const initialState: MapState = Map({
  // Not sure if you'll have to cast these or if TS will just pick them up as a compatible type
  layers: Map(),
  dateRange: Map(),
});
```

(haven't tested this, but something like it should work)

Reply:
Cf Issue #41